provider/provider.go

```diff
@@ -94,3 +94,17 @@ func getDefaultPath() string {
 		return "/usr/local/lib/summon"
 	}
 }
+
+//Create slice of all file names in the default path
+func GetAllProviders() []string{
+	files, err := ioutil.ReadDir(DefaultPath)
+    if err != nil {
+        fmt.Println(err.Error())
+		os.Exit(127)
```

Comment:
You normally don't want to `os.Exit()` in these methods where you're doing hard logic since that makes them much harder to unit test so I would probably return an error in that case and exit the program by returning the error to `runAction` which will do this for you.